lib/irb/command/ls.rb

```diff
@@ -20,27 +20,35 @@ class Ls < Base
           -g [query]  Filter the output with a query.
       HELP_MESSAGE
 
-      def self.transform_args(args)
-        if match = args&.match(/\A(?<args>.+\s|)(-g|-G)\s+(?<grep>[^\s]+)\s*\n\z/)
-          args = match[:args]
-          "#{args}#{',' unless args.chomp.empty?} grep: /#{match[:grep]}/"
+      def execute(arg)
+        if match = arg.match(/\A(?<args>.+\s|)(-g|-G)\s+(?<grep>.+)$/)
+          if match[:args].empty?
+            use_main = true
+          else
+            obj = @irb_context.workspace.binding.eval(match[:args])
```

Comment:
Nit: we can give the capture group with a clearer name:

```suggestion
        if match = arg.match(/\A(?<target>.+\s|)(-g|-G)\s+(?<grep>.+)$/)
          if match[:target].empty?
            use_main = true
          else
            obj = @irb_context.workspace.binding.eval(match[:target])
          end
```